cores/esp8266/umm_malloc/umm_performance.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * umm_malloc performance measurments and ESP specifics
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <pgmspace.h>
+#include <core_esp8266_features.h>
+#include "umm_performance.h"
+
+extern "C" {
+
+#ifdef UMM_CRITICAL_PERIOD_ANALYZE
+struct _UMM_TIME_STATS time_stats = {
+  {0xFFFFFFFF, 0U, 0U, 0U},
+  {0xFFFFFFFF, 0U, 0U, 0U},
+  {0xFFFFFFFF, 0U, 0U, 0U},
+  {0xFFFFFFFF, 0U, 0U, 0U} };
+
+bool ICACHE_FLASH_ATTR get_umm_get_perf_data(struct _UMM_TIME_STATS *p, size_t size) {
+    if (p && sizeof(time_stats) == size) {
+        uint32_t save_ps = xt_rsil(DEFAULT_CRITICAL_SECTION_INTLEVEL);
+        memcpy(p, &time_stats, size);
+        xt_wsr_ps(save_ps);
+        return true;
+    }
+    return false;
+}
+#endif
+
+#if defined(DEBUG_ESP_PORT) || defined(DEBUG_ESP_ISR)
+/*
+  Printing from the malloc routines is tricky. Since a lot of library calls
+  will want to do malloc.
+
+  Objective:  To be able to print "last gasp" diagnostic messages
+  when interrupts are disabled and w/o availability of heap resources.
+*/
+
+// ROM _putc1, ignores CRs and sends CR/LF for LF, newline.
+// Always returns character sent.
+int constexpr (*_rom_putc1)(int) = (int (*)(int))0x40001dcc;
```

Comment:
@earlephilhower what do you think of moving this to a .ld sceipt, like the others [here](https://github.com/esp8266/Arduino/blob/master/tools/sdk/ld/eagle.rom.addr.v6.ld)?

Reply:
A `PROVIDES` line in https://github.com/esp8266/Arduino/blob/master/tools/sdk/ld/eagle.rom.addr.v6.ld is probably the best place, yes.  There would also need to be a line in a header somewhere since this is an "undocumented" ROM function.

Reply:
We can do that in a separate PR.